From f50603723fd3c527aac99bf04707af65e584fb77 Mon Sep 17 00:00:00 2001
From: Yingren Wang <yingrenw@codeaurora.org>
Date: Tue, 8 Dec 2020 12:59:43 +0800
Subject: [PATCH 16/22] Launcher3: Fix gesture navigation fail to move to
 bottom

Go to the recents page from all apps page and rotate to horizontal, at
last switch to one app that only support portrait display, we will see
the gesture navigation always at device right and not move to bottom.
At this scenario, launcher state switch from overview to all apps, but
launcher not notify systemui update gesture navigation display.

To slove this, launcher must notify systemui update gesture navigation
display when launcher state switch from overview to all apps.

CRs-Fixed: 2828560

Change-Id: Ib2319045d3851b43dadd6a0e06ac65be2e5227f7
---
 .../src/com/android/quickstep/LauncherActivityInterface.java    | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/quickstep/src/com/android/quickstep/LauncherActivityInterface.java b/quickstep/src/com/android/quickstep/LauncherActivityInterface.java
index 0e0b02262b..a83c005f54 100644
--- a/quickstep/src/com/android/quickstep/LauncherActivityInterface.java
+++ b/quickstep/src/com/android/quickstep/LauncherActivityInterface.java
@@ -239,7 +239,7 @@ public final class LauncherActivityInterface extends
                     @Override
                     public void onStateTransitionComplete(LauncherState toState) {
                         // Are we going from Recents to Workspace?
-                        if (toState == LauncherState.NORMAL) {
+                        if (toState == LauncherState.NORMAL || toState == LauncherState.ALL_APPS) {
                             exitRunnable.run();
                             notifyRecentsOfOrientation(deviceState);
                             stateManager.removeStateListener(this);
-- 
2.34.1

